jms/src/main/scala/akka/stream/alpakka/jms/JmsSinkStage.scala

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.alpakka.jms
+
+import javax.jms.{JMSException, MessageProducer, TextMessage}
+
+import akka.Done
+import akka.stream.stage.{GraphStage, GraphStageLogic, InHandler, StageLogging}
+import akka.stream.{Attributes, Inlet, SinkShape}
+
+import scala.concurrent.Future
+
+final class JmsSinkStage(settings: JmsSettings) extends GraphStage[SinkShape[String]] {
+
+  private val in = Inlet[String]("JmsSink.in")
+
+  override def shape: SinkShape[String] = SinkShape.of(in)
+
+  override def createLogic(inheritedAttributes: Attributes): GraphStageLogic =
+    new GraphStageLogic(shape) with JmsConnector with StageLogging {
+
+      private var jmsProducer: MessageProducer = _
+
+      override private[jms] def jmsSettings = settings
+
+      private val initProducer = getAsyncCallback[MessageProducer](producer => {
+        jmsProducer = producer
+        pull(in)
+      })
+
+      private val askNext = getAsyncCallback[Done](_ => pull(in))
+
+      override private[jms] def onSessionOpened(): Unit =
+        jmsSession.createProducer().foreach { producer =>
+          initProducer.invoke(producer)
+        }
+
+      setHandler(in, new InHandler {
+        override def onPush(): Unit = {
+          val elem = grab(in)
+          Future {
```

Comment:
I've tried to implements and I realize that activemq was not implementing JMS 2. 
I have the feeling that the JMS 2 is not implemented a lot.
I could use the async way to send message with the risk to make the sink not usable for most of the users or downgrade JMS to 1.1 and keep the blocking way to send messages. 

What do you think ?